### PR TITLE
raft-log-engine: supply `prefill-for-recycle` configuration to enable starting engine in cold state.

### LINKED ISSUE #14371: [RaftLogEngine] Make log recycling effective when starting in cold by `prefill-for-recycle`.
## Development Task
In previous work, we've already introduced `enable-log-recycle` for reusing the obsolete raft logs for newly `append` entries. But this feature is not effective when starting with an empty db, only making sense when the count of logs up to threshold.

So, we want to introduce a new trigger, named with `prefill-for-recycle`, to enable this feature in advance when starting TiKV with an empty db, which has been done in https://github.com/tikv/raft-engine/pull/278.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: Close #14371 

What's Changed:

```commit-message
Adds a new configuration `raft-engine.prefill-for-recycle` for supporting to enable log recycling when starting TiKV in cold state.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
  - https://github.com/pingcap/docs-cn/pull/13339
  - https://github.com/pingcap/docs/pull/12826 
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Adds a new configuration `raft-engine.prefill-for-recycle` for supporting to enable log recycling when starting TiKV in cold state.
```

### CODE REVIEW

File: metrics/grafana/tikv_details.json [hunk 1]
```diff
@@ -45561,4 +45569,4 @@
   "title": "Test-Cluster-TiKV-Details",
   "uid": "RDVQiEzZz",
   "version": 1
-}
+}
```
Comment (tabokie) [inline metrics/grafana/tikv_details.json:45572]: revert

**Review by tabokie — COMMENTED** [review]
I didn't see config changes, I only see metrics update

**Review by tabokie — APPROVED** [review]

File: etc/config-template.toml [hunk 2]
```diff
@@ -1104,6 +1104,14 @@
 ## Default: false.
 # enable-log-recycle = false
 
+## Whether to prepare log files for recycling when start.
+## If `true`, batch empty log files will be prepared for recycling when
+## starting engine.
+## Only available for `enable-log-reycle` is true.
+##
+## Default: false
+# prefill-for-recycle = false
```
Comment (tonyxuqqi) [inline etc/config-template.toml:1113]: Should it have the test code to ensure the config passed to raft-engine correctly or behave correctly.

Comment (LykxSassinator) [reply etc/config-template.toml:1113]: Nope, it's unnecessary as these cases have already been covered by the related uts in `raft-engine` lib.

**Review by tonyxuqqi — APPROVED** [review]

**Pull request merged**